src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  mixPropTests,
+  propTests,
+} from '../../../../tests/playwright';
+import {
+  ButtonGroupForTest,
+  SelectedButtonGroupForTest,
+  ButtonGroupWithFeedbackIconForTest,
+  ButtonGroupWithLabelForTest,
+} from './ButtonGroup.story';
+
+test.describe('ButtonGroup', () => {
+  test.describe('base', () => {
+    test.describe('visual', () => {
+      [
+        ...propTests.defaultComponentPropTest,
+        ...propTests.blockPropTest,
+        ...propTests.disabledPropTest,
+        ...mixPropTests([
+          propTests.priorityPropTest,
+          propTests.sizePropTest,
+          propTests.disabledPropTest,
+        ]),
+      ].forEach(({
+        name,
+        onBeforeTest,
+        onBeforeSnapshot,
+        props,
+      }) => {
+        test(name, async ({
+          mount,
+          page,
+        }) => {
+          if (onBeforeTest) {
+            await onBeforeTest(page);
+          }
+
+          const component = await mount(<ButtonGroupForTest {...props} />);
+
+          if (onBeforeSnapshot) {
+            await onBeforeSnapshot(page, component);
+          }
+
+          const screenshot = await component.screenshot();
+          expect(screenshot).toMatchSnapshot();
+        });
+      });
+
+      test.describe('selectedButton', () => {
+        [
+          ...mixPropTests([
+            propTests.priorityPropTest,
+            propTests.disabledPropTest,
+          ]),
+        ].forEach(({
+          name,
+          onBeforeTest,
+          onBeforeSnapshot,
+          props,
+        }) => {
+          test(name, async ({
+            mount,
+            page,
+          }) => {
+            if (onBeforeTest) {
+              await onBeforeTest(page);
+            }
+
+            const component = await mount(<SelectedButtonGroupForTest {...props} />);
+
+            if (onBeforeSnapshot) {
+              await onBeforeSnapshot(page, component);
+            }
+
+            const screenshot = await component.screenshot();
+            expect(screenshot).toMatchSnapshot();
+          });
+        });
+      });
+
+      test.describe('buttonWithFeedbackIcon', () => {
+        [
+          ...mixPropTests([
+            propTests.priorityPropTest,
+            propTests.disabledPropTest,
+          ]),
+        ].forEach(({
+          name,
+          onBeforeTest,
+          onBeforeSnapshot,
+          props,
+        }) => {
+          test(name, async ({
+            mount,
+            page,
+          }) => {
+            if (onBeforeTest) {
+              await onBeforeTest(page);
+            }
+
+            const component = await mount(<ButtonGroupWithFeedbackIconForTest {...props} />);
+
+            if (onBeforeSnapshot) {
+              await onBeforeSnapshot(page, component);
+            }
+
+            const screenshot = await component.screenshot();
+            expect(screenshot).toMatchSnapshot();
+          });
+        });
+      });
+
+      test.describe('groupLabel', () => {
```

Comment:
On call with @lukasbriza, we decided to drop those composition-like tests due to its complexity. We decided to solve this in the future, if needed.